buildpacktoml.go

```diff
@@ -254,6 +260,33 @@ func (b *BuildpackTOML) readOutputFiles(bpLayersDir, bpPlanPath string, bpPlanIn
 	return br, nil
 }
 
+// we set default = true for web processes when platformAPI >= 0.6 and buildpackAPI < 0.6
+func updateDefaultProcesses(processes []launch.Process, bpAPI string, platformAPI string) {
+	if api.MustParse(platformAPI).Compare(api.MustParse("0.6")) < 0 || api.MustParse(bpAPI).Compare(api.MustParse("0.6")) >= 0 {
+		return
+	}
+
+	for i := range processes {
+		if processes[i].Type == "web" {
+			processes[i].Default = true
+		}
+	}
+}
```

Comment:
Could we move this logic into builder.go? Having `BuildpackTOML` care only about buildpack api things and `Builder` care (mostly) about platform api things gives a nice separation and is easier to test all the permutations. [Here](https://github.com/buildpacks/lifecycle/blob/fae1fe27c0acdfabe9538e3607e72af61c62ede9/builder.go#L104-L108) is an example of some post-processing that the `Builder` does depending on the platform api version.

Reply:
Good idea. Please see https://github.com/buildpacks/lifecycle/pull/520/commits/845f01c70137cd8ddf8c399a1b6a93f3199d6b72.